src/Chapter3.hs

```diff
@@ -373,6 +378,23 @@ after the fight. The battle has the following possible outcomes:
    doesn't earn any money and keeps what they had before.
 
 -}
+data LivingThing = MkLivingThing {
+  entityHealth :: Int,
+  entityAttack :: Int,
+  entityGold :: Int
+} deriving (Show)
+type Knight = LivingThing
+type Monster = LivingThing
+
+fight :: Monster -> Knight -> Int
```

Comment:
Nice usage of type aliases!
The only issue with type aliases and reusages of the same data type for Monster and Knight in here is that you can easily mix them when calling the function:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two types for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Entity
newtype Knight = Knight Entity
```